python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,163 @@
+
+        # Set region to match original region extent
+        gs.run_command(
+            "g.region",
+            n=self._extent["north"],
+            s=self._extent["south"],
+            e=self._extent["east"],
+            w=self._extent["west"],
+            env=self._vector_env,
+        )
+
+        # Get Center of tmp GRASS region
+        center = gs.parse_command("g.region", flags="cg", env=self._vector_env)
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+    def _convert_coordinates(self, x, y, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+        # Reformat input
+        coordinates = f"{x}, {y}"
+
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+
+        extent = {"north": north, "south": south, "east": east, "west": west}
+
+        return extent
+
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent["north"], extent["west"]], [extent["south"], extent["east"]]]
+
+    def add_vector(self, name):
+        """Imports vector into temporary WGS84 location,
+        re-formats to a GeoJSON and adds to folium map"""
+
+        # Find full name of vector
+        full_name = gs.find_file(name, element="vector")["fullname"]
+        name = gs.find_file(name, element="vector")["name"]
```

Comment:
To much overhead, this is calling g.findfile twice and twice parsing its output in Python. Maybe small overhead compared to time spend with manipulating the vector, but still not a good practice.

Naming that variable is hard, but the code is short (scope where the variable lives), so it can be something generic, like result (as in result from querying the spatial database) or some combination of file, vector, info, names, dict... I'm afraid there is no term for this thing in GRASS.